leshan-server-core/src/main/java/org/eclipse/leshan/server/security/SecurityInfo.java

```diff
@@ -50,14 +51,18 @@
     // X.509
     private final boolean useX509Cert;
 
+    // certificate usage
+    private final CertificateUsage certificateUsage;
+
     private SecurityInfo(String endpoint, String identity, byte[] preSharedKey, PublicKey rawPublicKey,
-            boolean useX509Cert) {
+            boolean useX509Cert, CertificateUsage certificateUsage) {
```

Comment:
I don't get what we need `Certificate Usage` in `SecurityInfo`.

SecurityInfo is information about to how device should connect to the server. (This is different than bootstrap configuration which define, the value to write on the device at bootstrap time)

E.g. client "C1" should connect using RPK with the  public key P1.

Here we just need to know it use x509 but I'm not sure server need to know which `CertificateUsage` is used.

Reply:
I am observing the code from my full set of changes so it should capture all usage so far.

In client side we have `org.eclipse.leshan.client.servers.ServerInfo#certificateUsage`. Which should already be used in client side for doing client side work.

Only call for using `org.eclipse.leshan.server.security.SecurityInfo#getCertificateUsage()` seems to be in `org.eclipse.leshan.server.redis.serialization.SecurityInfoSerDes#serialize()`.

So I have probably only captured that usage from Redis part -- but is the Redis code actually something that should be changed instead to something else? One option would use same JSON serialization that is used other servers. Probably the redis backend doesn't actually work in here as there is only one getter.

I am actually having a slight problem with Redis to understand the purpose of Redis in here -- it is often used for fast cache -- eg. simple key value pair thing. But now it seems that it also does configuration storing and not as secondary cache but as primarily storage?

Because of above I was thinking to use Redis as shared storage between bootstrap and server so that also bootsrapping configuration would be stored there (eg. use as device inventory). Eg. I was looking at option to handle certificate based device auto registration in my EST support demo code.

Reply:
I don't get your point at all :confused: . 
I read my comment above again and see there is a typo and so maybe you misunderstood my question, please read it again.

My current feeling is that we don't need to add `Certificate Usage` to  `SecurityInfo`, we don't need it to identify a Client at server side. 

> I am actually having a slight problem with Redis to understand the purpose of Redis in here -- it is often used for fast cache -- eg. simple key value pair thing. But now it seems that it also does configuration storing and not as secondary cache but as primarily storage?

We use it as a store not a cache. (and this is an optional module which allow us to test persistent store use case)